service-coordtransform/src/main/java/fi/nls/oskari/PointTransformerImpl.java

```diff
@@ -0,0 +1,160 @@
+
+import com.vividsolutions.jts.geom.Coordinate;
+import fi.nls.oskari.domain.geo.Point;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.util.PropertyUtil;
+import fi.nls.paikkatietoikkuna.coordtransform.CoordTransQueryBuilder;
+import fi.nls.paikkatietoikkuna.coordtransform.CoordTransWorker;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PointTransformerImpl extends NLSFIPointTransformer {
+
+    private static final Logger log = LogFactory.getLogger(PointTransformerImpl.class);
+    private static final String PROP_END_POINT = "coordtransform.endpoint";
+    private static final List<String> UNSUPPORTED = Arrays.asList("EPSG:4326");
+    private static final String EPSG_ETRS89_GEOGRAPHIC = "EPSG:4258";
+    private static final double GK_ZONE_OFF_MINUTUES = 30;
+    private static final int EPSG_CODE_GK19FIN = 3873;
+    private static final String PREFIX_EPSG = "EPSG:";
+    private static final String NLSFI_GKn = "NLSFI:etrs_gk";
+    private static final Map<String, String> EPSG_CODE_MAP = initEpsgCodeMap();
+    private static final Map<String, Boolean> AXIS_ORDER_MAP = initAxisOrderMap();
+    private CoordTransWorker worker;
+    private String endPoint;
+
+    private static Map<String, String> initEpsgCodeMap() {
+        Map<String,String> codeMap = new HashMap<>();
+        codeMap.put("LATLON:kkj", PREFIX_EPSG + "4123");
+        codeMap.put("NLSFI:ykj", PREFIX_EPSG + "2393");
+        return codeMap;
+    }
+
+    private static Map<String, Boolean> initAxisOrderMap() {
+        Map<String,Boolean> axisIsNorthFirstMap = new HashMap<>();
+        axisIsNorthFirstMap.put(PREFIX_EPSG + "4123", true);
+        return axisIsNorthFirstMap;
+    }
+
+    @Override
+    public Point reproject(Point point, String sourceSRS, String targetSRS) {
+        if (point == null || sourceSRS == null || targetSRS == null) {
+            return null;
+        }
+        try {
+            if (sourceSRS.equals(NLSFI_GKn)) {
+                int zone = (int)(point.getLon() / 1E6);
+                sourceSRS = getEPSGForGKnFIN(getGKnFINZone(zone));
+            }
+            String sourceEPSG = getEpsgCode(sourceSRS);
+
+            if (targetSRS.equals(NLSFI_GKn)) {
+                targetSRS = getEPSGForGKnFIN(getGKnFINZone(point, sourceEPSG));
+            }
+            String targetEPSG = getEpsgCode(targetSRS);
+
+            if (isSupported(sourceEPSG, targetEPSG)) {
+                try {
+                    CoordTransQueryBuilder queryBuilder = getQueryBuilder(sourceEPSG, targetEPSG);
+                    Coordinate coord = getCoordinate(point, isNorthAxisFirst(sourceEPSG));
+                    worker.transform(queryBuilder, Arrays.asList(coord));
+                    return getPoint(coord, isNorthAxisFirst(targetEPSG));
+                } catch (Exception e) {
```

Comment:
Since we are doing a catch'em all here I think we can just use the single try/catch instead of a nested ones to make this more readable.